src/gsy_e/models/market/forward.py

```diff
@@ -0,0 +1,138 @@
+    def _get_market_slot_duration(current_time: DateTime, _config) -> duration:
+        """Return duration of market slots inside the market block."""
+
+    def create_future_market_slots(self, current_market_time_slot: DateTime,
+                                   config: "SimulationConfig") -> None:
+        if not ConstSettings.ForwardMarketSettings.ENABLE_FORWARD_MARKETS:
+            return
+        self._create_future_market_slots(self._get_start_time(current_market_time_slot),
+                                         self._get_end_time(current_market_time_slot), config)
+
+
+class DayForwardMarket(ForwardMarketBase):
+    """Day forward market implementation"""
+
+    @staticmethod
+    def _get_start_time(current_time: DateTime) -> DateTime:
+        return current_time.set(hour=0, minute=0).add(days=1)
+
+    @staticmethod
+    def _get_end_time(current_time: DateTime) -> DateTime:
+        return current_time.set(hour=0, minute=0).add(weeks=1, days=1, hours=-1)
+
+    @staticmethod
+    def _get_market_slot_duration(_current_time: DateTime, _config) -> duration:
+        return duration(hours=1)
+
+
+class WeekForwardMarket(ForwardMarketBase):
+    """Week forward market implementation"""
+
+    @staticmethod
+    def _get_start_time(current_time: DateTime) -> DateTime:
+        days_until_next_monday = 7 - (current_time.day_of_week - 1)
+        return current_time.set(hour=0, minute=0).add(days=days_until_next_monday)
+
+    @staticmethod
+    def _get_end_time(current_time: DateTime) -> DateTime:
+        return current_time.set(hour=0, minute=0).add(years=1)
+
+    @staticmethod
+    def _get_market_slot_duration(_current_time: DateTime, _config) -> duration:
+        return duration(weeks=1)
+
+
+class MonthForwardMarket(ForwardMarketBase):
+    """Month forward market implementation"""
+
+    @staticmethod
+    def _get_start_time(current_time: DateTime) -> DateTime:
+        return current_time.set(day=1, hour=0, minute=0).add(months=1)
+
+    @staticmethod
+    def _get_end_time(current_time: DateTime) -> DateTime:
+        return current_time.set(day=1, hour=0, minute=0).add(years=2)
+
+    @staticmethod
+    def _get_market_slot_duration(current_time: DateTime, _config) -> duration:
+        number_of_days_month = monthrange(current_time.year, current_time.month)[1]
+        return duration(days=number_of_days_month)
+
+
+class YearForwardMarket(ForwardMarketBase):
+    """Year forward market implementation"""
+
+    @staticmethod
+    def _get_start_time(current_time: DateTime) -> DateTime:
+        return current_time.set(month=1, day=1, hour=0, minute=0).add(years=2)
+
+    @staticmethod
+    def _get_end_time(current_time: DateTime) -> DateTime:
+        return current_time.set(month=1, day=1, hour=0, minute=0).add(years=6)
+
+    @staticmethod
+    def _get_market_slot_duration(current_time: DateTime, _config) -> duration:
+        return duration(years=1, days=1) if current_time.is_leap_year() else duration(years=1)
```

Comment:
The DateTime object is absolutely able to handle leap years on its own. There's definitely no need for us to handle them or the result will be unexpected:

![Screenshot from 2022-07-13 08-05-33](https://user-images.githubusercontent.com/31365000/178662449-a3f3b2fa-3eeb-43be-a61d-a2231db9693c.png)